epm/Distribution/Client/IndexUtils.hs

```diff
@@ -450,7 +451,7 @@ packageIndexFromCache mkPkg hnd entrs mode = accum mempty [] entrs
       -- from the index tarball if it turns out that we need it.
       -- Most of the time we only need the package id.
       ~(pkg, pkgtxt) <- unsafeInterleaveIO $ do
-        mPatch <- patchedPackageCabalFile pkgid
+        mPatch <- patchedPackageCabalFile pkgid defaultPatchesDir
```

Comment:
> Well I think the Build Type decides how to do the configure step so they need to know it up front even before extracting. My ignorance of this fact caused many frustrations.

After giving more thought on this, I think we can add the same option to `epm update` command. If `--patches-directory` is given, we can also copy the entire patches directory to `~/.epm/patches` apart from updating the package list.